### ISSUE TITLE
New room list, sub lists have started rendering more room tiles than are actually visible

### ISSUE THREAD

**Issue by t3chguy** [vector-im/riot-web#14426]
This causes havoc with keyboard navigation in the room list, it looks like the calculated padding doesn't always account for the showN button

**Comment by t3chguy** [comment]
![image](https://user-images.githubusercontent.com/2403652/87241849-ec418000-c41e-11ea-83a0-32926883f8e8.png)

https://github.com/matrix-org/matrix-react-sdk/blob/213e2df9fced040e4e143dfc32259abecfb9cc33/src/components/views/rooms/RoomSublist2.tsx#L138-L144
looks to be at fault but I cannot digest how to resolve it right now

**Comment by turt2live** [comment]
This got changed from floor to ceiling for other changes, and may very well be at fault: https://github.com/matrix-org/matrix-react-sdk/blob/213e2df9fced040e4e143dfc32259abecfb9cc33/src/components/views/rooms/RoomSublist2.tsx#L157

**Comment by t3chguy** [comment]
It is not, I had 1.5 visible but it said 3.

The issue was it was not accounting for the showN button even though it was taking up space in the height which it was measuring

**Comment by turt2live** [comment]
right, I think that might still be the ceiling and long-term layout issues. Changing it back to `floor`, `mx_RoomListLayoutStore.resetLayouts()`, and reloading might fix it.

**Comment by t3chguy** [comment]
I doubt it given that `layout.visibleTiles` is `3` because of https://github.com/matrix-org/matrix-react-sdk/blob/213e2df9fced040e4e143dfc32259abecfb9cc33/src/components/views/rooms/RoomSublist2.tsx#L206-L209 where the padding offset is too small due to it not including the showN and thus it thinking there's a whole extra tile showing than there is

**Comment by turt2live** [comment]
Yay.
